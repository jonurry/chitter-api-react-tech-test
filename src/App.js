import React, { Component } from 'react';
import './css/App.css';
import Peeps from './Peeps.js';

class App extends Component {
  constructor(props) {
    super(props);
    this.api = props.api;
  }
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1>Chitter Tech Test</h1>
        </header>
        <Peeps api={this.api} />
      </div>
    );
  }
}

export default App;
