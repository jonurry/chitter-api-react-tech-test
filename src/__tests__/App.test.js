import React from 'react';
import { waitForElement, render } from 'react-testing-library';
import App from '../App';
import Api from '../api';

jest.mock('../api');

describe('<App>', () => {
  it('renders without crashing', () => {
    render(<App api={new Api()} />);
  });

  test('App Matches Snapshot', async () => {
    const { container, getByText } = await render(<App api={new Api()} />);
    await waitForElement(() => getByText('my first peep :)'));
    expect(container.firstChild).toMatchSnapshot();
  });
});
