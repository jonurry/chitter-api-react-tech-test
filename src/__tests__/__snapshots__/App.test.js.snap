// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App> App Matches Snapshot 1`] = `
<div
  class="App"
>
  <header
    class="App-header"
  >
    <h1>
      Chitter Tech Test
    </h1>
  </header>
  <form>
    <label>
      Peep:
      <input
        name="peep"
        type="text"
      />
    </label>
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
  <div
    class="peeps"
  >
    <div
      class="peep"
    >
      <div
        class="peep-text"
        data-testid="peep-text"
      >
        my first peep :)
      </div>
      <div
        class="peep-name-date"
        data-testid="peep-name-date"
      >
        <span
          class="peeper"
        >
          kay
        </span>
         on
         
        <span
          class="peep-date"
        >
          23-06-2018 14:21
        </span>
      </div>
      <button
        data-testid="like-button"
      >
        Likes: 
        1
      </button>
      <button>
        Delete
      </button>
    </div>
    <div
      class="peep"
    >
      <div
        class="peep-text"
        data-testid="peep-text"
      >
        another peep
      </div>
      <div
        class="peep-name-date"
        data-testid="peep-name-date"
      >
        <span
          class="peeper"
        >
          jon
        </span>
         on
         
        <span
          class="peep-date"
        >
          1-07-2018 09:06
        </span>
      </div>
      <button
        data-testid="like-button"
      >
        Likes: 
        2
      </button>
      
    </div>
    <div
      class="peep"
    >
      <div
        class="peep-text"
        data-testid="peep-text"
      >
        and another peep
      </div>
      <div
        class="peep-name-date"
        data-testid="peep-name-date"
      >
        <span
          class="peeper"
        >
          kay
        </span>
         on
         
        <span
          class="peep-date"
        >
          2-07-2018 23:56
        </span>
      </div>
      <button
        data-testid="like-button"
      >
        Likes: 
        2
      </button>
      <button>
        Delete
      </button>
    </div>
  </div>
</div>
`;
